### ISSUE TITLE
TS UI

### ISSUE THREAD

**Issue by x-a-n-a-x** [OpenRA/OpenRA#7474]
In order to help developers get rid of the old UI code https://github.com/OpenRA/OpenRA/pull/7461, I have started a draft, a first attempt at making a TS UI. It may or may not be be final UI (I don't know yet), but we'll see.

Current branch:
https://github.com/x-a-n-a-x/OpenRA/commits/new-TS-UI

As you can see, it's far from being finished and there are still bugs with some coordinates in chrome.yaml or ingame-player.yaml files. But little by little, I try to fix problems. 

![screen shot 02-14-15 at 08 25 pm](https://cloud.githubusercontent.com/assets/1732773/6200789/cda2f5a4-b487-11e4-863d-36ba1283a52d.PNG)

**Comment by x-a-n-a-x** [comment]
Branch updated.
(now using Tomsons26's TS logos)

TODO:
- fix the vertical powerbar not being correctly repeated
- fix these (order buttons) symbols overlapping the production icons
- fix the scroll up and scroll down buttons (and use the other artwork for these buttons)
- align Order and Production buttons correctly
- rework the artwork behind the Timer, Cash & Power values
- change font color for Timer, Cash & Power? or add a 1px drop down shadow?

**Comment by Phrohdoh** [comment]
We shouldn't have a powerbar and a digital readout.

**Comment by x-a-n-a-x** [comment]
That's what was planned for the D2K UI.. both..

**Comment by penev92** [comment]
Oh, sweet logo.
I don't care about the number indicator. We **need** the power bar!
But there would probably be some time between the UI getting merged and someone making the power bar work (and not just be a placeholder), so both should be OK for now.

**Comment by Mailaender** [comment]
I don't think you will be able to fix the radar. It is one of the things on the bucket list https://github.com/OpenRA/OpenRA/issues/5350 as the minimap is not yet adapted properly to the isometric perspective. Just make the Nod logo a bit smaller to compensate for now.

**Comment by Mailaender** [comment]
This is looking great (except for the d2k build palette icons being all over the place)!

**Comment by pchote** [comment]
I'm not a fan of using the GDI sidebar with a Nod logo for Nod.  We will need to build a proper nod themed version too.

**Comment by Mailaender** [comment]
Yes, but not now.

**Comment by pchote** [comment]
:-1: to that. History shows that it will then never happen.

**Comment by Mailaender** [comment]
Whatever. This builds upon https://github.com/OpenRA/OpenRA/pull/7461 so that can go first anyway.

**Comment by x-a-n-a-x** [comment]
Branch updated (once more):
![screen shot 02-15-15 at 12 24 pm](https://cloud.githubusercontent.com/assets/1732773/6203114/73605fe0-b50f-11e4-9222-4b065f64dcc9.PNG)

**Comment by Mailaender** [comment]
Nice. Only the aircraft queue left to do and the trike symbol needs to replaced by a tank.

**Comment by Phrohdoh** [comment]
There is also some empty space by the scroll buttons, I wonder what we could add there to reuse the portion of the screen occupied by the UI.

**Comment by x-a-n-a-x** [comment]
> Only the aircraft and defense queue left to do and the trike symbol needs to replaced by a tank.

Are these Production buttons required? I haven't enabled them yet but everything is ready (in chrome.png).

> There is also some empty space by the scroll buttons, I wonder what we could add there to reuse the portion of the screen occupied by the UI.

I'll see what I can do with this empty space.

Here is what I've been working on during the last hours: 
![chrome](https://cloud.githubusercontent.com/assets/1732773/6204301/3bc9e434-b546-11e4-9ee4-5bf3dab941f8.png)

**Comment by penev92** [comment]
Nice :+1: 
What is that cobra thing?

**Comment by Phrohdoh** [comment]
I believe that is a shp made by the Nod artist for TI from ppm.

I cannot remember how to spell his name.

**Comment by dan9550** [comment]
@Phrohdoh why are we ditching the power bar D:? EDIT: Sorry miss read, power bar all the way IMO.

I also think at the moment things look a little cluttered, but i would make my mind up about that once i saw it in game.

Overall :+1:

**Comment by x-a-n-a-x** [comment]
Thanks to @pchote, I have been able to fix the Order buttons at the top (they were GDI before).

![screen shot 02-16-15 at 09 45 pm](https://cloud.githubusercontent.com/assets/1732773/6219101/c3c2feea-b625-11e4-9c90-06620296f99f.PNG)

**Comment by Mailaender** [comment]
:cool: That dark design really improves the mood of playing another faction.

**Comment by dan9550** [comment]
i think we need something behind the nod logo in the radar box

**Comment by x-a-n-a-x** [comment]
Little by little, it's getting closer to completion (the vertical powerbars on this screenshot is just a mockup, the rest is already in my branch).

![screen shot 02-17-15 at 10 19 pm](https://cloud.githubusercontent.com/assets/1732773/6237906/fcbd3bda-b6f6-11e4-9007-95769b210820.PNG)

- [ ] fix the scroll up and scroll down buttons (they don't change when clicked)
- [ ] observer UI not done yet.. (openra crashes if you win a game and creates a huge debug.log file)
- [ ] the NOD UI still uses the GDI order symbols
- [ ] chrome.yaml features production-icons: chrome.png, production-icons-nod: chrome.png, production-icons-gdi: chrome.png (same for order-icons: chrome.png, order-icons-nod: chrome.png and order-icons-gdi: chrome.png). I am not sure all these paragraphs are useful.

**Comment by Phrohdoh** [comment]
I would like to see the vertical midline of the scroll triangle graphics line up with the vertical lines separating the palette columns.

**Comment by x-a-n-a-x** [comment]
heh, not really a problem. Added on my todo list.

**Comment by dan9550** [comment]
The top of the interface where the power and cash is looks a bit cluttered, could easily be fixed with some extra padding in the UI.
I would also be interested in seeing the scroll buttons aligned the the right of the UI.

**Comment by Mailaender** [comment]
Now that https://github.com/OpenRA/OpenRA/pull/7461 is merged, you can start preparing a pull request.

**Comment by x-a-n-a-x** [comment]
My branch is still broken (and not all remaining tasks have been completed, check my TODO list above). https://github.com/x-a-n-a-x/OpenRA/commits/new-TS-UI

Current error: Exception of type `System.ArgumentException`: Sprite / was not found.

**Comment by Mailaender** [comment]
I made some comments on https://github.com/x-a-n-a-x/OpenRA/commit/1e5b8cbf1a06a5b596b1cb53c90fb02e43152a42.

> move all units from Vehicles to Tanks

Why? Just replace the Dune trike icon with something more generic.

**Comment by Phrohdoh** [comment]
Because the original had all vehicles/tanks in one queue, so @x-a-n-a-x please do that.

**Comment by Mailaender** [comment]
I filed https://github.com/OpenRA/OpenRA/pull/7574 as @x-a-n-a-x seems to need help again.

**Comment by x-a-n-a-x** [comment]
Update : I have retried to fix the errors I had for the TS UI and I think it's a success. 
If you want, you can test https://github.com/x-a-n-a-x/OpenRA/tree/new-TS-build-palette-skin

I don't consider the UI ready yet, I would like to tweak it further.

**Comment by x-a-n-a-x** [comment]
I have updated my branch once again. I think this UI is done. 
I need testers, reviewers. Thanks.

Just an observation : when you click on the Diplomacy button, it disables the Debug, Diplomacy and Options buttons.

**Comment by Mailaender** [comment]
Can you send a pull request?

**Comment by x-a-n-a-x** [comment]
Do you want me to squash commits first ?

**Comment by Mailaender** [comment]
Yes, please.

**Comment by x-a-n-a-x** [comment]
It will take longer than expected. I have big problems trying to squash commits.

**Comment by Mailaender** [comment]
Squashing is never a problem. You are probably facing rebase conflicts, because leaving this unfinished for over a months left your branch and bleed diverge too much.

**Comment by x-a-n-a-x** [comment]
PR : https://github.com/OpenRA/OpenRA/pull/7699
